src/components/button/Button.tsx

```diff
@@ -5,95 +5,133 @@ import { TTheme } from '../../foundations/theme-provider/themes/theme.types';
 import { setAlpha, blendColors, Utils } from '../../shared';
 
 import ButtonBase from './Button.base';
-import { DEFAULT_BUTTON_SIZE, DEFAULT_BUTTON_VARIANT } from './Button.constants';
+import {
+    BUTTON_ICON_MARGIN_MAP,
+    DEFAULT_BUTTON_SIZE,
+    DEFAULT_BUTTON_VARIANT,
+} from './Button.constants';
 import { PButton } from './Button.props';
 
 const getButtonVariables = ({
-    theme: { palette, action, text },
+    theme: { palette, action, background },
     onClick,
-    disabled = false,
     destructive = false,
+    inverted = false,
+    disabled = false,
     size = DEFAULT_BUTTON_SIZE,
     variant = DEFAULT_BUTTON_VARIANT,
 }: ThemedStyledProps<PButton, TTheme>): FlattenSimpleInterpolation => {
     const isDisabled = disabled || !Utils.isFunction(onClick);
+    const isInverted = inverted && !destructive;
+
+    // get default opacities for 'primary' or 'inverted primnary' buttons
+    const opacities: Record<string, number> = {
+        background: isDisabled ? 0.08 : 1,
+        hover: isInverted ? 0.08 : 0.16,
+        active: isInverted ? 0.16 : 0.32,
+    };
+
+    const colors: Record<string, string> = {
+        main: destructive ? palette.alert.main : palette.primary.main,
+        text: destructive ? palette.alert.contrast : palette.primary.contrast,
+        action: action.hover,
+        // border will be set afterwards, since it is influenced by other props
+        border: '',
+    };
+
+    // handle inverted here before re-assigning colors
+    if (inverted && !destructive) {
+        colors.main = background.contrast;
+    }
 
-    let mainColor = destructive ? palette.alert.main : palette.primary.main;
-    let hoverColor = action.hover;
-    let bgOpacity = isDisabled ? 0.08 : 1;
-
-    let textColor = text.contrast;
-    let borderColor = mainColor;
-
-    if (variant !== 'primary') {
-        // for variants `secondary` and `tertiary` use the
-        // primary color, but set alpha to `0`
-        bgOpacity = 0;
-        // hoverColor is set to the primary button color
-        // for non-`primary` buttons
-        hoverColor = mainColor;
-        textColor = mainColor;
+    colors.border = colors.main;
+
+    switch (variant) {
+        case 'tertiary':
+            opacities.background = 0.04;
+            opacities.hover = 0.08;
+            opacities.active = 0.12;
+
+            colors.text = colors.main;
+            colors.action = colors.main;
+            break;
+        case 'secondary':
+            opacities.background = 0;
+            opacities.hover = 0.04;
+            opacities.active = 0.08;
+
+            colors.text = colors.main;
+            colors.action = colors.main;
+            break;
+        case 'primary':
+        default:
+            // 'inverted primary' buttons have different colors for text and actions
```

Comment:
Oh maybe it's a bit different to how I imagine it being, if we can have inverted primary buttons. Still if there will be a way to unify these I am all for it.

Reply:
mmh, it is implemented as stated in the current design ... we will see if this is valid for future iterations though!